src/mon/Monitor.cc

```diff
@@ -2103,16 +2104,25 @@ void Monitor::handle_probe_reply(MonOpRequestRef op)
              << " vs my version " << paxos->get_version()
              << " (ok)"
              << dendl;
-
-    if (monmap->contains(name) &&
-        !monmap->get_addrs(name).front().is_blank_ip()) {
+    bool in_map = false;
+    const auto& my_info = monmap->mon_info.find(name);
```

Comment:
This looks like undefined behaviour, my_info will end up as a ref to a free'd temporary since find returns an interator by value?